### PR TITLE
Introduce Option.ifPresent as Option.forEach alternative but with a more symmetric name to existing Option.ifAbsent

### PR DESCRIPTION
**Pull request by balomia**

### CODE REVIEW

**Comment by Synesso** [comment]
You'll need to run `gradle apiDump` and commit the result. You can also consider adding a note to CHANGELOG

File: lib/src/main/kotlin/app/cash/quiver/extensions/Option.kt [hunk 1]
```diff
@@ -20,11 +20,11 @@ inline fun <T> Option<T>.ifAbsent(f: () -> Unit): Unit {
...
 /**
- * Turns your Option into a Validated list of T if it's a Some.
- * If it's a None, will return a Nel of the error function passed in
+ * Runs a side effect if the option is a Some. Same as forEach, just a different name for better symmetry with ifAbsent.
  */
-inline fun <T, E> Option<T>.toValidatedNel(error: () -> E): ValidatedNel<E, T> =
-  ValidatedNel.fromOption(this) { nonEmptyListOf(error()) }
+inline fun <A> Option<A>.ifPresent(f: (A) -> Unit) {
+  onSome { f(it) }
+}
```
Comment (cwmyers) [inline lib/src/main/kotlin/app/cash/quiver/extensions/Option.kt:27]: Nit: Given this is just the implementation of `forEach` I think this should just return `forEach`

File: lib/src/test/kotlin/app/cash/quiver/extensions/OptionTest.kt [hunk 2]
```diff
@@ -14,6 +14,7 @@ import io.kotest.property.arrow.core.option
 import io.kotest.property.checkAll
 import app.cash.quiver.extensions.traverse as quiverTraverse
 import app.cash.quiver.extensions.traverseEither as quiverTraverseEither
+import app.cash.quiver.extensions.ifPresent as quiverIfPresent
```
Comment (cwmyers) [inline lib/src/test/kotlin/app/cash/quiver/extensions/OptionTest.kt:17]: Do we need to rename it?

File: lib/src/main/kotlin/app/cash/quiver/extensions/Option.kt (hunk 1 above)
Comment (balomia) [reply lib/src/main/kotlin/app/cash/quiver/extensions/Option.kt:27]: truuu

**Review by Synesso — APPROVED** [review]
Noice

**Review by cwmyers — APPROVED** [review]
Nailed it

**Pull request merged**